src/irr/asset/IImageLoader.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2009-2012 AnastaZIuk
+// This file is part of the "Irrlicht Engine".
+// For conditions of distribution and use, see copyright notice in irrlicht.h
+
+#ifndef __IRR_I_IMAGE_LOADER_H_INCLUDED__
+#define __IRR_I_IMAGE_LOADER_H_INCLUDED__
+
+#include "irr/core/core.h"
+
+#include "IWriteFile.h"
+#include "irr/asset/IAsset.h"
+
+namespace irr
+{
+	namespace asset
+	{
+
+		class IImageLoader : public virtual core::IReferenceCounted
+		{
+			public:
+
+				IImageLoader();
+				virtual ~IImageLoader();
```

Comment:
do a pure virtual destructor, do the constructor as a default

Reply:
pure virtual destructor, why?

Reply:
pure virtual destructor is for a class that we never ever intend to instantiate (Abstract Data Type) and that is a pure interface (no members, etc.)

Reply:
but I need to provide a definition of the destructor, where should I put it?

Reply:
I need to create for it cpp file however

Reply:
`virtual ~IImageLoader() =0;` thats a pure virtual destructor, makes the class abstract, the destructor never gets instantiated, so you never need the definition

Reply:
`IReferenceCounted` has one